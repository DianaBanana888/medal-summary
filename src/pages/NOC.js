import React from 'react'
import info from '../jsonInfo';
import Table from 'react-bootstrap/Table';

export default function NOC() {
  const sport = info.NOCMedals
  return (
    <div className="table-medal-first">
      <p>NOC Table</p>
      <Table striped bordered hover>
        <thead>
          <tr>
            <th>NOC Team</th>
            <th className="gold">Golden</th>
            <th className="silver">Silver</th>
            <th className="bronze">Bronze</th>
            <th>Total</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{sport.NOC.c_Name}</td>
            <td>{sport.Medals.n_Gold}</td>
            <td>{sport.Medals.n_Silver}</td>
            <td>{sport.Medals.n_Bronze}</td>
            <td>{sport.Medals.n_Total}</td>
          </tr>
        </tbody>
      </Table>
    </div>
  )
}
