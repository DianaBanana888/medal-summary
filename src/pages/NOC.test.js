import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NOC from './NOC';

test('renders speed skating button', () => {

  render(<NOC />);
  const teamNetherlands = screen.getByText(/Netherlands/i);
  expect(teamNetherlands).toBeInTheDocument();
});

it('renders NOC correctly', () => {
  const tree = renderer
    .create(<NOC />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
