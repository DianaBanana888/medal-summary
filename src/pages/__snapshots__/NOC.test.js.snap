// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NOC correctly 1`] = `
<div
  className="table-medal-first"
>
  <p>
    NOC Table
  </p>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          NOC Team
        </th>
        <th
          className="gold"
        >
          Golden
        </th>
        <th
          className="silver"
        >
          Silver
        </th>
        <th
          className="bronze"
        >
          Bronze
        </th>
        <th>
          Total
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Netherlands
        </td>
        <td>
          8
        </td>
        <td>
          7
        </td>
        <td>
          9
        </td>
        <td>
          24
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
